Packet++/header/SmtpLayer.h

```diff
@@ -0,0 +1,341 @@
+			/// Receiver indication
+			RCPT = ('R') | ('C' << 8) | ('P' << 16) | ('T' << 24),
+			/// Abort transaction
+			RSET = ('R') | ('S' << 8) | ('E' << 16) | ('T' << 24),
+			/// Identify user
+			VRFY = ('V') | ('R' << 8) | ('F' << 16) | ('Y' << 24),
+			/// Start TLS handshake
+			STARTTLS = (('S') | ('T' << 8) | ('A' << 16) | ('R' << 24) |
+					static_cast<uint64_t>(('T') | ('T' << 8) | ('L' << 16) | ('S' << 24)) << 32),
+			/// Reverse the role of sender and receiver
+			TURN = ('T') | ('U' << 8) | ('R' << 16) | ('N' << 24),
+			/// Send mail to terminal
+			SEND = ('S') | ('E' << 8) | ('N' << 16) | ('D' << 24),
+			/// Send mail to terminal or to mailbox
+			SOML = ('S') | ('O' << 8) | ('M' << 16) | ('L' << 24),
+			/// Send mail to terminal and mailbox
+			SAML = ('S') | ('A' << 8) | ('M' << 16) | ('L' << 24),
+			/// Authenticate client and server
+			AUTH = ('A') | ('U' << 8) | ('T' << 16) | ('H' << 24),
+			/// Reverse the role of sender and receiver
+			ATRN = ('A') | ('T' << 8) | ('R' << 16) | ('N' << 24),
+			/// Submit mail contents
+			BDAT = ('B') | ('D' << 8) | ('A' << 16) | ('T' << 24),
+			/// Request to start SMTP queue processing
+			ETRN = ('E') | ('T' << 8) | ('R' << 16) | ('N' << 24),
+			/// Release status of the channel
+			XADR = ('X') | ('A' << 8) | ('D' << 16) | ('R' << 24),
+			/// Release status of the circuit checking facility
+			XCIR = ('X') | ('C' << 8) | ('I' << 16) | ('R' << 24),
+			/// Release status of the number of messages in channel queues
+			XSTA = ('X') | ('S' << 8) | ('T' << 16) | ('A' << 24),
+			/// Release status of whether a compiled configuration and character set are in use
+			XGEN = ('X') | ('G' << 8) | ('E' << 16) | ('N' << 24)
+		};
+
+		/** A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		SmtpRequestLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+			: SmtpLayer(data, dataLen, prevLayer, packet){};
+
+		/**
+		 * A constructor that creates layer with provided input values
+		 * @param[in] command SMTP command
+		 * @param[in] option Argument of the command
+		 */
+		explicit SmtpRequestLayer(const SmtpCommand &command, const std::string &option = "")
+			: SmtpLayer(getCommandAsString(command), option){};
+
+		/**
+		 * Set the command of request message
+		 * @param[in] code Value to set command
+		 * @return True if the operation is successful, false otherwise
+		 */
+		bool setCommand(SmtpCommand code);
+
+		/**
+		 * Get the command of request message
+		 * @return SmtpCommand Value of the command
+		 */
+		SmtpCommand getCommand() const;
+
+		/**
+		 * Get the command of request message as string
+		 * @return std::string Value of the command as string
```

Comment:
In the `@return` doxygen document we usually don't put the type, so `std::string` is redundant.
The same for the rest of the methods in this file